source-mysql/discovery.go

```diff
@@ -171,6 +171,15 @@ func (db *mysqlDatabase) translateRecordFields(columnTypes map[string]interface{
 		return nil
 	}
 	for id, val := range f {
+		// MariaDB versions 10.4 and up include a synthetic `DB_ROW_HASH_1` column in the
+		// binlog row change events for certain tables with unique hash indices (see issue
+		// https://github.com/estuary/connectors/issues/1344). In such cases we won't have
+		// any type information for the column, but we also don't want it anyway, so as a
+		// special case we just delete the 'DB_ROW_HASH_1' property from the document.
+		if id == "DB_ROW_HASH_1" && columnTypes[id] == nil {
```

Comment:
Is it possible to somehow end up actually having `columnTypes[id]` here?

Reply:
I haven't tried but I suspect some supported versions of MySQL (most likely older ones and/or non-MariaDB) may prrmit a user to create a column by that name.

In such a case we would want to still capture that value, though really this is probably just theoretical and nobody will ever have an actual column by that name.